cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/consumption/AwsEFSConsumptionCalculator.java

```diff
@@ -0,0 +1,117 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.consumption;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.validation.ValidationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.amazonaws.services.elasticfilesystem.model.DescribeFileSystemsResult;
+import com.amazonaws.services.elasticfilesystem.model.FileSystemDescription;
+import com.cloudera.thunderhead.service.metering.events.MeteringEventsProto;
+import com.sequenceiq.cloudbreak.cloud.ConsumptionCalculator;
+import com.sequenceiq.cloudbreak.cloud.aws.common.connector.resource.AwsEfsCommonService;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudConsumption;
+import com.sequenceiq.cloudbreak.cloud.model.StorageSizeRequest;
+import com.sequenceiq.cloudbreak.cloud.model.StorageSizeResponse;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.cloudbreak.common.mappable.StorageType;
+import com.sequenceiq.common.model.FileSystemType;
+
+@Service
+public class AwsEFSConsumptionCalculator implements ConsumptionCalculator {
+
+    private static final StorageType EFS = StorageType.EFS;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsEFSConsumptionCalculator.class);
+
+    private static final long NO_BYTE_IN_MB = 1000L * 1000L;
+
+    private static final int DATE_RANGE_WIDTH_IN_DAYS = 2;
+
+    @Inject
+    private AwsEfsCommonService awsEfsCommonService;
+
+    @Override
+    public void validate(CloudConsumption cloudConsumption) throws ValidationException {
+        String storageLocation = cloudConsumption.getStorageLocation();
+        if (storageLocation == null || !storageLocation.startsWith("fs-")) {
+            throw new ValidationException(String.format("EFS id must start with 'fs-' if required file system type is '%s'!",
+                    FileSystemType.EFS.name()));
+        }
+    }
+
+    @Override
+    public StorageSizeResponse calculate(StorageSizeRequest request) {
+        DescribeFileSystemsResult result = awsEfsCommonService.getEfsSize(
+                request.getCredential(),
+                request.getRegion().value(),
+                request.getStartTime(),
+                request.getEndTime(),
+                request.getObjectStoragePath());
+        Optional<FileSystemDescription> latestFileSystemDescription = result.getFileSystems().stream().findFirst();
```

Comment:
Wouldn't this actually return the earliest FileSystemDescription? I'm assuming the intention is to get the most up to data, by getting the latest description, which in this case should be the last one, or if the descriptions are not in order for some reason, then the list should be searched for the latest description.

Reply:
in case of this request you can get multiple EFS information but currently we are only requesting one at a time

Reply:
thank you for the clarification!